app/js/player/controllers/PlayerCtrl.js

```diff
@@ -0,0 +1,138 @@
+"use strict";
+/**
+ * Controller for thisissoon FM player view, handles searching
+ * the spotify api for songs using ngMaterial autocomplete and
+ * angular-spotify
+ * @module FM.player.PlayerCtrl
+ * @author SOON_
+ */
+angular.module("FM.player.PlayerCtrl",[
+    "FM.api.PlayerTransportResource",
+    "FM.api.PlayerMuteResource",
+    "FM.api.PlayerVolumeResource",
+    "FM.api.ERRORS"
+])
+/**
+ * @class PlayerCtrl
+ * @param {Object}  $scope
+ * @param {Factory} PlayerTransportResource
+ * @param {Factory} PlayerMuteResource
+ * @param {Factory} PlayerVolumeResource
+ * @param {Object}  ERRORS
+ */
+.controller("PlayerCtrl", [
+    "$scope",
+    "$q",
+    "PlayerTransportResource",
+    "PlayerMuteResource",
+    "PlayerVolumeResource",
+    "ERRORS",
+    function ($scope, $q, PlayerTransportResource, PlayerMuteResource, PlayerVolumeResource) {
+
+        /**
+         * The currently playing track
+         * @property track
+         * @type     {Object}
+         */
+        $scope.track = {};
+
+        /**
+         * Tracks the state of mute
+         * @property mute
+         * @type     {Boolean}
+         */
+        $scope.mute = false;
+
+        /**
+         * @property volume
+         * @type     {Integer}
+         */
+        $scope.volume = 0;
+
+        /**
+         * Get all the data
+         * @method getAllData
+         */
+        $scope.getAllData = function getAllData(){
+            $q.all([
+                PlayerTransportResource.get().$promise,
+                PlayerMuteResource.get().$promise,
+                PlayerVolumeResource.get().$promise
+            ]).then(function (response){
+                $scope.track = response[0];
+                $scope.mute = response[1].mute;
+                $scope.volume = response[2].volume;
+            });
+        };
+
+        /**
+         * Set paused state and send request to API
+         * @method resume
+         */
+        $scope.resume = function resume() {
+            PlayerTransportResource.resume().$promise
+                .then($scope.onSuccess);
+        };
+
+        /**
+         * Set paused state and send request to API
+         * @method pause
+         */
+        $scope.pause = function pause() {
+            PlayerTransportResource.pause().$promise
+                .then($scope.onSuccess);
+        };
+
+        /**
+         * Send DELETE request to API
+         * @method skip
+         */
+        $scope.skip = function skip() {
+            PlayerTransportResource.skip().$promise
+                .then($scope.onSuccess);
+        };
+
+        /**
+         * send POST request to update volume
+         * @method skip
+         */
+        $scope.updateVol = function updateVol() {
+            PlayerVolumeResource.save({ volume: $scope.volume }).$promise
+                .then($scope.onSuccess);
```

Comment:
yes we do but since error message currently don't work anyway, was planning to do so once error message are put back in. Also was curious why there was error handing originally in the `success` handlers, does the FM api return 400 status codes in a 200?